### DISCUSSION TITLE
kernel compile failed for Netgear R7000

### DISCUSSION THREAD

**Discussion by kd1995-max** [Ysurac/openmptcprouter#3152] [Q&A]
hello community,
I am trying to build an image for Netgear R7000 
but getting error in 
```target/linux failed to build.```

below is the log
```
mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6250-squashfs.chk.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6250-squashfs.chk
mkchkimg:      Kernel Len: 34865152
mkchkimg: Kernel Checksum: 0xa91a5ff8
mkchkimg:  Image Checksum: 0xa91a5ff8
mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6300-v2-squashfs.chk.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6300-v2-squashfs.chk
mkchkimg:        Board Id: U12H315T30_NETGEAR
mkchkimg:          Region: World Wide (WW)
mkchkimg:      Kernel Len: 34865152
mkchkimg: Kernel Checksum: 0xda8cfc27
mkchkimg:  Image Checksum: 0xda8cfc27
mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r7900-squashfs.chk.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r7900-squashfs.chk
mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-buffalo_wzr-900dhp-squashfs.factory-DHP-EU.bin.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-buffalo_wzr-900dhp-squashfs.factory-DHP-EU.bin
make[5]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/target/linux/bcm53xx/image'
make[4]: *** [Makefile:30: install] Error 2
make[4]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/target/linux/bcm53xx'
make[3]: *** [Makefile:11: install] Error 2
make[3]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/target/linux'
time: target/linux/install#79.70#23.03#49.06
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:25: target/linux/install] Error 1
make[2]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source'
make[1]: *** [target/Makefile:19: /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/staging_dir/target-arm_cortex-a9_musl_eabi/stamp/.target_install] Error 2
make[1]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source'
make: *** [/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/include/toplevel.mk:230: world] Error 2
```
please help to build.

**Comment by a834003831** [comment]
> hello community, I am trying to build an image for Netgear R7000 but getting error in `target/linux failed to build.`
> 
> below is the log
> 
> ```
> mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6250-squashfs.chk.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6250-squashfs.chk
> mkchkimg:      Kernel Len: 34865152
> mkchkimg: Kernel Checksum: 0xa91a5ff8
> mkchkimg:  Image Checksum: 0xa91a5ff8
> mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6300-v2-squashfs.chk.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r6300-v2-squashfs.chk
> mkchkimg:        Board Id: U12H315T30_NETGEAR
> mkchkimg:          Region: World Wide (WW)
> mkchkimg:      Kernel Len: 34865152
> mkchkimg: Kernel Checksum: 0xda8cfc27
> mkchkimg:  Image Checksum: 0xda8cfc27
> mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r7900-squashfs.chk.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-netgear_r7900-squashfs.chk
> mv /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-buffalo_wzr-900dhp-squashfs.factory-DHP-EU.bin.new /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/build_dir/target-arm_cortex-a9_musl_eabi/linux-bcm53xx_generic/tmp/openmptcprouter-v0.60beta1-master-d303c1bc-r0+16862-170d9e447d-bcm53xx-generic-buffalo_wzr-900dhp-squashfs.factory-DHP-EU.bin
> make[5]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/target/linux/bcm53xx/image'
> make[4]: *** [Makefile:30: install] Error 2
> make[4]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/target/linux/bcm53xx'
> make[3]: *** [Makefile:11: install] Error 2
> make[3]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/target/linux'
> time: target/linux/install#79.70#23.03#49.06
>     ERROR: target/linux failed to build.
> make[2]: *** [target/Makefile:25: target/linux/install] Error 1
> make[2]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source'
> make[1]: *** [target/Makefile:19: /home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/staging_dir/target-arm_cortex-a9_musl_eabi/stamp/.target_install] Error 2
> make[1]: Leaving directory '/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source'
> make: *** [/home/krishanu/netgear-r7000/openmptcprouter/myplatform/5.4/source/include/toplevel.mk:230: world] Error 2
> ```
> 
> please help to build.

Hello, I also have this device. May I ask if you have already compiled it into firmware? I would like to flash OMR onto this device as well, but I don’t know how to compile it.
